aptos-move/move-examples/data_structures/sources/smart_table.move

```diff
@@ -0,0 +1,569 @@
+                level = level + 1;
+            };
+        };
+        let i = 1;
+        while (i <= 256) {
+            assert!(*borrow(&table, i) == i, 0);
+            i = i + 1;
+        };
+        assert!(table.num_buckets == 257, table.num_buckets);
+        assert!(load_factor(&table) == 99, 0);
+        assert!(length(&table) == 256, 0);
+        destroy(table);
+    }
+
+    #[test]
+    fun smart_table_update_configs() {
+        let table = new();
+        let i = 0;
+        while (i < 200) {
+            add(&mut table, i, i);
+            i = i + 1;
+        };
+        assert!(length(&table) == 200, 0);
+        update_target_bucket_size(&mut table, 10);
+        update_split_load_threshold(&mut table, 50);
+        while (i < 400) {
+            add(&mut table, i, i);
+            i = i + 1;
+        };
+        assert!(length(&table) == 400, 0);
+        i = 0;
+        while (i < 400) {
+            assert!(contains(&table, i), 0);
+            assert!(remove(&mut table, i) == i, 0);
+            i = i + 1;
+        };
+        destroy_empty(table);
+    }
+
+    #[test_only]
+    fun make(): SmartTable<u64, u64> {
+        let table = new_with_config<u64, u64>(0, 50, 10);
+        let i = 0u64;
+        while (i < 100) {
+            add(&mut table, i, i);
+            i = i + 1;
+        };
+        table
+    }
+
+    #[test]
+    fun test_for_each() {
+        let t = make();
+        let s = 0;
+        for_each(t, |x, y| {
+            s = s + x + y;
+        });
+        assert!(s == 9900, 0)
+    }
+
+    #[test]
+    fun test_for_each_ref() {
+        let t = make();
+        let s = 0;
+        for_each_ref(&t, |x, y| {
+            s = s + *x + *y;
+        });
+        assert!(s == 9900, 0);
+        destroy(t);
+    }
+
+    #[test]
+    fun test_for_each_mut() {
+        let t = make();
+        for_each_mut(&mut t, |_key, val| {
+            let val : &mut u64 = val;
+            *val = *val + 1
+        });
+        for_each_ref(&t, |key, val| {
+            assert!(*key + 1 == *val, *key);
+        });
+        destroy(t);
+    }
+
+    #[test]
+    fun test_fold() {
+        let t = make();
+        let r = fold(t, 1, |accu, key, val| {
+            accu + key + val
+        });
+        assert!(r == 9901, 0);
+    }
+
+        #[test]
+        fun test_map() {
+            let m = make();
+            let r = map(m, |val| val + 1);
+            for_each_ref(&r, |key, val| {
+                assert!(*key + 1 == *val, *key);
+            });
+            destroy(r);
+        }
+
+        #[test]
+        fun test_filter() {
+            let m = make();
+            let r = filter(m, |val| *val < 50);
+            assert!(length(&r) == 50, 0);
+            assert!(
+                fold(r, 0, |accu, _key, val| {
+                    accu + val
+                }) == 1225, 1);
+        }
+
+        #[test]
+        fun test_any() {
+            let t = make();
+            let r = any(&t, |_k, v| *v >= 99);
```

Comment:
This line triggers:
```
thread 'test_data_structures' has overflowed its stack
fatal runtime error: stack overflow
```
Do you have any idea what's wrong with the implementation of `any`?
@movekevin

Reply:
test_map_ref, test_any and test_all do not work for the same reason.

Reply:
You must use compiler/cli build with `--release`. Its not related to inline functions, they are just hitting existing problems in the functional programming design of the Move compiler.

Reply:
I delete `test_all` for now cuz it always triggers a bug.